### ISSUE TITLE
Installation needs root access?

### ISSUE THREAD

**Issue by stoeps13** [HCL-TECH-SOFTWARE/connections-automation#17]
When I check the `become `statements which don't point to a `become_user`, I assume the setup needs a `root` user?

**Comment by dejo1307** [comment]
Yes, that's true. Check the playbook for particular section, most of them do have become set to yes in playbooks already, which defines then what it expects in tasks.

**Comment by stoeps13** [comment]
I don't think become is enough here, because additionally all checks for installed software needs to be changed to <home of install user>/var/ibm

**Comment by stoeps13** [comment]
@sabrina-yee I think this is a really important topic. Most of our Enterprise customer do not allow installing as `root`, so even my workaround until now to run a script in the end to move the user to something non-root is not possible. Even doing sudo in such environment is not allowed. Until now we deploy manually or with handcrafted bash scripts, so I know it is possible.

These customers provide OpenShift or central Kubernetes environments, so the setup of K8s is not possible or needed. It would be really great when all the WebSphere tasks can run as non-root (this is best practice for security). I know that some config can only be made with root, but this can be defined as requirement.

Regards
Christoph

**Comment by sabrina-yee** [comment]
Hi Christoph, if I understand correctly, you're referring to this [instructions](https://help.hcltechsw.com/connections/v7/admin/install/t_non-root_install.html) to manually install as non-root?  The main reason it's not automated yet is more about which part of the repo we're investing resource to develop/maintain wrt our deployment needs.  

Beyond that, I'd imagine in the future to invest in things like getting the repo to use the Ansible modular playbooks and roles as you suggested a while ago so the WebSphere part development is more in as-needed basis.

Thanks,
Sabrina

**Comment by stoeps13** [comment]
you're right, if I can select then the modular approach is more important, because then it is easier to build exceptions for these environments.
